Topics/jupyter-notebook.md

```diff
@@ -0,0 +1,154 @@
+5 minutes.
+
+
+# Index
+ - [Why Jupyter Notebooks](#Why-Jupyter-Notebooks)
+ - [Learn to use it](#Learn-to-use-it)
+ - [Considerations](#Considerations)
+ - [Tips and Tricks](#Tips-and-Tricks)
+ - [Notebooks Examples](#Notebooks-Examples)
+ - [Conclusions](#Conclusions)
+
+
+Let's dive right in!
+ 
+### Why Jupyter Notebooks
+
+[Jupyter notebooks](https://jupyter.org/) are a way of embedding ready-to-execute python code with markdown documentation, images, and other such things, all in one file, as well as a program for presenting and running them. 
+
+They're nice if you want to demonstrate a short script to another person, particularly in a data science context, because of some built-in plotting niceties and the fact that you can run cells of code individually. 
+
+They are [widely used in Data Science](https://www.nature.com/articles/d41586-018-07196-1), because as we know the Data Science Process is an experimental and incremental process.
+
+There are several reasons to choose Jupyter Notebook for the first phases of the Data Science process (e.g. data exploration, data cleaning):
+
+- it's easy to write more than one line of the code before running. You do not need an editor on the side, crafting functions and then copying to the shell
+
+- results presentation is richer than console. It supports graphs, images, tables. Effectively anything that can be represented by html, css and js
+
+- if you make a mistake, you can trivially rerun that "cell" (chunk of code, nothing else) once you fix the mistake
+
+- you can install it on some more powerful machine and interact with it in your browser, so works for you if you're not super friendly with terminal stuff
+
+- you can export run results as html page. a run of a notebook is effectively a static thing. notebook consists of code + results of running the code.
+
+_Curiosity:_ Project Jupyter's name is a reference to the three core programming languages supported by Jupyter, which are Julia, Python and R, and also a homage to Galileo's notebooks recording the discovery of the moons of Jupiter.
+
+### Learn to use it
+
+Now that you understand why it is important to use Jupyter Notebooks, let's see how to actually learn how to use them.
+
+The guide for beginners that Virgilio recommends is as follows:
+
+- [**Jupyter Notebook for Beginners Tutorial**](https://www.dataquest.io/blog/jupyter-notebook-tutorial/)
+
+You will:
+
+- Cover the basics of installing Jupyter and creating your first notebook
+- Delve deeper and learn all the important terminology
+- Explore how easily notebooks can be shared and published online. Indeed, the tutorial _is a Jupyter Notebook!_ Everything here was written in the Jupyter Notebook environment, though you are viewing it in a read-only form.
+
+Once you're done with this tutorial, you _have the option_ to dive deeper and follow this one:
+
+- [**Jupyter Notebook for Advanced Users Tutorial**](https://www.dataquest.io/blog/advanced-jupyter-notebooks-tutorial/)
```

Comment:
There is a [nice tutorial](https://www.dataquest.io/blog/advanced-jupyter-notebooks-tutorial/) talking about additional things you can do in a Jupyter notebook

Reply:
it's the same I linked..